sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java

```diff
@@ -246,26 +248,54 @@ public void setSessionFactory(SessionFactory sessionFactory) {
-    String str = DbUtils.convertStringToSearchQuery(externalName, false);
-    Criteria criteria = currentSession().createCriteria(SampleIdentityImpl.class);
-    criteria.add(Restrictions.or(Restrictions.ilike("externalName", str), Restrictions.ilike("alias", str)));
+  public Collection<SampleIdentity> getIdentitiesByExternalNameOrAliasPartialMatch(String externalNames) throws IOException {
+    if (isStringEmptyOrNull(externalNames)) return Collections.emptySet();
     @SuppressWarnings("unchecked")
-    Collection<SampleIdentity> records = criteria.list();
+    List<SampleIdentity> records = (List<SampleIdentity>) SampleIdentityImpl.getSetFromString(externalNames)
+        .stream().map(extName -> {
+          String str = DbUtils.convertStringToSearchQuery(extName, false);
+          Criteria criteria = currentSession().createCriteria(SampleIdentityImpl.class);
+          criteria.add(Restrictions.or(Restrictions.ilike("externalName", str), Restrictions.ilike("alias", str)));
+          return criteria.list();
+        }).flatMap(list -> list.stream()).collect(Collectors.toList());
     return records;
   }
 
   @Override
-  public Collection<SampleIdentity> getIdentitiesByExternalNameAndProject(String externalName, Long projectId) throws IOException {
-    if (isStringEmptyOrNull(externalName)) return Collections.emptySet();
+  public Collection<SampleIdentity> getIdentitiesByExactExternalName(String externalNames) throws IOException {
+    if (externalNames == null) return Collections.emptyList();
+
+    Collection<SampleIdentity> partialMatches = getIdentitiesByExternalNameOrAliasPartialMatch(externalNames);
+    return filterOnlyExactExternalNameMatches(partialMatches, externalNames);
+  }
+
+  @Override
+  public List<SampleIdentity> getIdentitiesByExactExternalNameAndProject(String externalNames, Long projectId) throws IOException {
+    if (isStringEmptyOrNull(externalNames)) return Collections.emptyList();
     if (projectId == null) throw new IllegalArgumentException("Must provide a projectId in search");
-    Criteria criteria = currentSession().createCriteria(SampleIdentityImpl.class);
-    criteria.add(Restrictions.eq("project.id", projectId));
-    criteria.add(Restrictions.eq("externalName", externalName).ignoreCase());
     @SuppressWarnings("unchecked")
-    Collection<SampleIdentity> records = criteria.list();
-    return records;
+    List<SampleIdentity> records = (List<SampleIdentity>) SampleIdentityImpl.getSetFromString(externalNames)
+        .stream().map(extName -> {
+          String str = DbUtils.convertStringToSearchQuery(extName, false);
+          Criteria criteria = currentSession().createCriteria(SampleIdentityImpl.class);
+          criteria.add(Restrictions.eq("project.id", projectId));
+          criteria.add(Restrictions.ilike("externalName", str));
+          return criteria.list();
+        }).flatMap(list -> list.stream()).collect(Collectors.toList());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)